v2/dtos/reading.go

```diff
@@ -65,3 +67,37 @@ func ToReadingModel(r BaseReading, device string) models.Reading {
 	}
 	return readingModel
 }
+
+func FromReadingModelToDTO(reading models.Reading) BaseReading {
+	var baseReading BaseReading
+	switch reading.(type) {
```

Comment:
use 
`switch r := reading.(type)`
Then, you don't need to cast in each case
https://tour.golang.org/methods/16

Reply:
Modified.